### ISSUE TITLE
download converted flow visualizations

### ISSUE THREAD

**Issue by elsdvlee** [TiboStr/dataflow-visualization#50]
I would like to be able to download the converted flow visualizations as png, svg or animated gif so it can be easily integrated in presentations and deliverables. It would also be nice to have the option to adapt the image size when downloading.
Similar to the actions in  [https://mermaid.live/](https://mermaid.live/).

**Comment by TiboStr** [comment]
**For future reference**

Capturing a part of a screen is relatively new (https://w3c.github.io/mediacapture-region/).

The demo (https://github.com/w3c/mediacapture-region/tree/main/demo) does not work yet for me.

--------------------


There is an easy-to-use React hook to capture your screen (https://github.com/tylerbuchea/use-screen-recording), but you can only capture your entire screen. I don't see how this is more useful than third-party solutions that let you capture a part of your screen, such as [peek](https://github.com/phw/peek), which I am using.

**Comment by TiboStr** [comment]
@elsdvlee, could you take a look at https://knowledgeonwebscale.github.io/dataflow-visualization/#/recording and https://knowledgeonwebscale.github.io/dataflow-visualization/#/recording2.

Do you want me to implement the screenshot and/or capture feature? Personally, I think this is not more useful than using third-party software to capture your screen as webm/gif or use the key combination provided by your OS to take a screenshot.

But if you still want these features, I'll do my best to implement them.

**Comment by TiboStr** [comment]
There are libraries that let you screenshot a React component, but that does nor work in combination with ReactFlow. Something like this is the result:
![img](https://user-images.githubusercontent.com/71405687/186868027-bb58d169-7f75-4f68-97a7-7ba4210a61f9.jpg)

So I think the only way to do it is by doing it like https://knowledgeonwebscale.github.io/dataflow-visualization/#/recording2.

**Comment by elsdvlee** [comment]
@TiboStr Indeed, this is not more useful than using third-party software to capture your screen as webm/gif or use the key combination provided by your OS to take a screenshot. I was hoping for a Mermaid-like solution (https://mermaid.live/) where you can set a higher resolution for more complicated images.  Maybe not close this issue yet, but focus first on other issues.
